creature.cpp

```diff
@@ -1144,6 +1144,8 @@ static string getAttackParam(AttackType type) {
     case AttackType::SPELL: return "spell";
     case AttackType::POSSESS: return "touch";
   }
+  assert("__FUNCTION__ case unknown attack type");
+  return "";
```

Comment:
I hate this, even more boilerplate. I use clang during development and it can detect that the switch exhausts all possibilities and doesn't print warnings.